src/control/ToolHandler.cpp

```diff
@@ -495,15 +541,58 @@ auto ToolHandler::isSinglePageTool(ToolPointer toolPointer) -> bool {
            toolType == TOOL_FLOATING_TOOLBOX || toolType == TOOL_DRAW_SPLINE;
 }
 
-auto ToolHandler::getToolPointer(ToolPointer toolpointer) -> Tool* {
-    switch (toolpointer) {
-        case ToolPointer::current:
-            return this->currentTool;
-        case ToolPointer::toolbar:
+
+auto ToolHandler::getSelectedTool(SelectedTool selectedTool) -> Tool* {
+    switch (selectedTool) {
+        case SelectedTool::active:
+            return this->activeTool;
+        case SelectedTool::toolbar:
             return this->toolbarSelectedTool;
-        case ToolPointer::button:
-            return this->buttonSelectedTool;
         default:
-            g_error("This ToolPointer does not exist.");
+            g_error("This button does not exist.");
+    }
+}
+
+auto ToolHandler::getButtonTool(Buttons button) -> Tool* {
+    switch (button) {
+        case Buttons::BUTTON_ERASER:
+            return this->eraserButtonTool.get();
+        case Buttons::BUTTON_STYLUS:
+            return this->stylusButton1Tool.get();
+        case Buttons::BUTTON_STYLUS2:
+            return this->stylusButton2Tool.get();
+        case Buttons::BUTTON_MIDDLE:
+            return this->mouseMiddleButtonTool.get();
+        case Buttons::BUTTON_RIGHT:
+            return this->mouseRightButtonTool.get();
+        case Buttons::BUTTON_TOUCH:
+            return this->touchDrawingButtonTool.get();
+        default:
+            g_error("This button is not defined for ToolHandler.");
+    }
+}
+
+void ToolHandler::resetButtonTool(Tool* tool, Buttons button) {
+    switch (button) {
+        case Buttons::BUTTON_ERASER:
+            this->eraserButtonTool.reset(new Tool(tool));
+            break;
+        case Buttons::BUTTON_STYLUS:
+            this->stylusButton1Tool.reset(new Tool(tool));
+            break;
+        case Buttons::BUTTON_STYLUS2:
+            this->stylusButton2Tool.reset(new Tool(tool));
+            break;
+        case Buttons::BUTTON_MIDDLE:
+            this->mouseMiddleButtonTool.reset(new Tool(tool));
+            break;
+        case Buttons::BUTTON_RIGHT:
+            this->mouseRightButtonTool.reset(new Tool(tool));
+            break;
+        case Buttons::BUTTON_TOUCH:
+            this->touchDrawingButtonTool.reset(new Tool(tool));
+            break;
+        default:
+            g_error("This button is not defined for ToolHandler.");
     }
 }
```

Comment:
Please always add a newline at the end of a file. Nowadays C-Compilers don't really require it anymore but it is still a good thing to do for compatibility. You never know what toolchain somebody tries to use. And it removes that ugly sign...

Reply:
Well, not coming from the cpp world I can definitely add this to my TIL, thanks :pray: 

In case someone else stumbles upon this: This is how you set it in vscode https://stackoverflow.com/a/54270768.

Reply:
This was added in 38d509654dd5b017f84bf8b17f59b0a907d83638 and will be automatically applied in every code I touch :innocent: